pvanalytics/system.py

```diff
@@ -401,3 +403,172 @@ def infer_orientation_daily_peak(power_or_poa, sunny, tilts,
                 best_azimuth = azimuth
                 best_tilt = tilt
     return best_azimuth, best_tilt
+
+
+def _power_residuals_from_clearsky(system_params,
+                                   ghi, dhi, dni,
+                                   solar_zenith, solar_azimuth,
+                                   power_ac, temperature,
+                                   wind_speed,
+                                   temperature_coefficient,
+                                   temperature_model_parameters):
+    # Return the residuals between a system with parameters given in `params`
+    # and the data in `power_ac`.
+    #
+    # Parameters
+    # ----------
+    # system_params : array-like
+    #     array of four floats: tilt, azimuth, DC capacity, and inverter
+    #     DC input limit.
+    # ghi : Series
+    #     Clear sky GHI
+    # dhi : Series
+    #     Clear sky DHI
+    # dni : Series
+    #     Clear sky DNI
+    # solar_zenith : Series
+    #     Solar zenith at the same times as data in `power_ac`
+    # solar_azimuth : Series
+    #     Solar azimuth at the same times as data in `power_ac`
+    # power_ac : Series
+    #     Measured AC power under clear sky conditions.
+    # temperature : float or Series
+    #     Air temperature at which to model the hypothetical system. If a
+    #     float then a constant temperature is used. If a Series, must have
+    #     the same index as `power_ac`. [C]
+    # wind_speed : float or Series
+    #     Wind speed. If a float then a constant wind speed is used. If a
+    #     Series, must have the same index as `power_ac`. [m/s]
+    # temperature_model_parameters : dict
+    #     Parameters fot the cell temperature model.
+    #
+    # Returns
+    # -------
+    # Series
+    #     Difference between `power_ac` and the PVWatts output with the
+    #     given parameters.
+    tilt = system_params[0]
+    azimuth = system_params[1]
+    dc_capacity = system_params[2]
+    dc_limit = system_params[3]
+    poa = pvlib.irradiance.get_total_irradiance(
+        tilt, azimuth,
+        solar_zenith,
+        solar_azimuth,
+        dni, ghi, dhi
+    )
+    temp_cell = pvlib.temperature.sapm_cell(
+        poa['poa_global'],
+        temperature,
+        wind_speed,
+        **temperature_model_parameters
+    )
+    pdc = pvlib.pvsystem.pvwatts_dc(
+        poa['poa_global'],
+        temp_cell,
+        dc_capacity,
+        temperature_coefficient
+    )
+    return power_ac - pvlib.inverter.pvwatts(pdc, dc_limit)
+
+
+def _rsquared(data, residuals):
+    # Calculate the coefficient of determination from `residuals`
+    model = data + residuals
+    _, _, r, _, _ = scipy.stats.linregress(model, data)
+    return r*r
+
+
+def orientation_fit_pvwatts(power_ac, ghi, dhi, dni,
+                            solar_zenith, solar_azimuth,
+                            temperature=25, wind_speed=0,
+                            temperature_coefficient=-0.002,
```

Comment:
I think I just pulled that from the documentation for `pvlib.pvsystem.pvwatts_dc()`. Do you have a recommendation for a better default, I don't really have enough familiarity to say what a good default would be here?

Reply:
that docstring says "Typically -0.002 to -0.005 per degree C." Maybe use the average?

Reply:
Since PVWatts doesn't use a linear temperature model, I'd suggest using -0.004 as the default. Would be interesting to calculate the slope of a linear approximation to the Fuentes model with PVWatts defaults. Someday.